<a href="https://colab.research.google.com/github/KOMBOU12/Marius/blob/main/Matrix_Completion.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Le but est d'implémenter le code présent dans ce lien avec un jeu de donnée de mon choix : https://colab.research.google.com/drive/1-KStBJ_gHRjRrAwaMj6cBZLymqiEjpnm?usp=sharing

In [1]:
pip install matrix-completion

In [ ]:
import numpy as np
from matrix_completion import svt_solve, calc_unobserved_rmse

U = np.random.randn(20, 5)
V = np.random.randn(15, 5)
R = np.random.randn(20, 15) + np.dot(U, V.T)

mask = np.round(np.random.rand(20, 15))
R_hat = svt_solve(R, mask)

print("RMSE:", calc_unobserved_rmse(U, V, R_hat, mask))

RMSE: 2.3708963533613394


Le code génère une matrice de rang faible R avec des valeurs manquantes, utilise l'algorithme **SVT pour imputer les valeurs manquantes**, et évalue la qualité de l'imputation en calculant la RMSE sur les valeurs non observées.

In [ ]:
import numpy as np
from matrix_completion import svt_solve, calc_unobserved_rmse

# Generation de nouvelles matrices U, V, et R
U_new = np.random.randn(25, 7)  # U est maintenant 25x7
V_new = np.random.randn(18, 7)  # V est maintenant 18x7
R_new = np.random.randn(25, 18) + np.dot(U_new, V_new.T)  # R est maintenant 25x18 avec une faible structure de rang

# Creation d'un nouveau masque pour les valeurs manquantes
mask_new = np.round(np.random.rand(25, 18))

# Application de la SVT pour compléter la matrice
R_hat_new = svt_solve(R_new, mask_new)

# Calcul du RMSE pour les valeurs pas observées
rmse_new = calc_unobserved_rmse(U_new, V_new, R_hat_new, mask_new)

# le nouveau résultat du RMSE
print("New RMSE:", rmse_new)


New RMSE: 1.930425374949002


Le **RMSE de 1.93** signifie que le modèle fait une erreur moyenne de 1.93 sur les prédictions pour les valeurs manquantes.